app/views/staff/activities/show.html.haml

```diff
@@ -16,8 +16,9 @@
       .govuk-grid-column-two-thirds
         %h2.govuk-heading-m
           = t("page_content.organisation.programmes")
-        = form_tag(organisation_fund_programmes_path(@activity.organisation, @activity), method: "post") do
-          = submit_tag t("page_content.organisation.button.create_programme"), class: "govuk-button"
+        - if current_user.administrator? || current_user.fund_manager?
```

Comment:
I wonder if we can add the concept of `level` into the Pundit policies so that when we ask say `policy(@activity).create?` the `create?` check will take the level into account. 

```
# activity_policy.rb
def create?
  if record.fund? && (current_user.administrator? || current_user.fund_manager?) ||
     record.programme? && (current_user.administrator? || current_user.fund_manager?) ||
     record.project? && (current_user.administrator? || current_user.delivery_partner?)  
```

This would benefit us by keeping our views clean and our authorisation logic DRY with great test coverage.

Reply:
So this is where I stumble on Pundit stuff, because if you're checking if a user can create an Activity, you don't yet know what level the Activity is at. If you pry the Activity at this point, you get something like this:

`Activity(id: uuid, organisation_id: uuid, created_at: datetime, updated_at: datetime, identifier: string, sector: string, title: string, description: text, status: string, planned_start_date: date, planned_end_date: date, actual_start_date: date, actual_end_date: date, recipient_region: string, flow: string, finance: string, aid_type: string, tied_status: string, wizard_status: string, level: string, activity_id: uuid, funding_organisation_name: string, funding_organisation_reference: string, funding_organisation_type: string, accountable_organisation_name: string, accountable_organisation_reference: string, accountable_organisation_type: string, extending_organisation_id: uuid)`

There's no way of knowing what level it's at because it's not been created yet.

Reply:
That makes sense. Hmm. I'll have another go and see if I can think of another way to make this less reptitive.

Reply:
When I check out this branch this logic isn't in this view so I'm finding it hard to get myself into the scenario to test this.

Reply:
It's an outdated commit - this logic is now in `app/views/staff/shared/activities/_activity_activities.html.haml:6`